### DISCUSSION TITLE
Unrestricted types, only checking member extensions

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/csharplang#8370] [General]
be similar to
`arr[ .. ]`only need `Count`and`Slice`.
`foreach` only need `GetEnumerator`,and no matter what this method returns, as long as the return value has`Current `,` MoveNext `.

Many engines have a vector. If I write extension methods for them, only for this type.
I'm not sure if using interface bridging will cause boxing or any other performance issues.
I think it would be great if the extension method written could be used for any type with X and Y members.

**Comment by HaloFour** [comment]
I believe that would fall under extension implementation: https://github.com/dotnet/csharplang/issues/5497

I expect that it would require at least specifying the extension, although perhaps the compiler can automatically wire up the compatible members.
